channel.go

```diff
@@ -103,8 +103,14 @@ func (chManager *channelManager) reconnect() error {
 		return err
 	}
 
-	chManager.channel.Close()
-	chManager.connection.Close()
+	err = chManager.channel.Close()
```

Comment:
I don't think this is the way to handle this. If there's an issue with the connection this method will return an error - but what do we want to do in that case? Well, we would want to reconnect which is what we're doing. However, by forwarding along the error, we won't use the reconnected, new channel